src/main/java/com/celements/course/registration/RegistrationData.java

```diff
@@ -81,8 +84,21 @@ public void setPrice(int price) {
     this.price = price;
   }
 
-  public Optional<String> getValidationKey() {
-    return Optional.ofNullable(validationKey).filter(not(String::isEmpty));
+  public static final String generateNewValidationKey() {
+    return RandomStringUtils.randomAlphanumeric(24);
+  }
```

Comment:
I'm ok with this being a static method, but perhaps it's the wrong place here. how about putting it into `CourseParticipantClass` since it is a default to the `FIELD_VALIDATION_KEY`?

Reply:
👍